package.json

```diff
@@ -85,7 +85,8 @@
     "storage-memory": "0.0.2"
   },
   "peerDependencies": {
-    "redux": ">4.0.0"
+    "redux": ">4.0.0",
+    "@react-native-community/async-storage": "^1.2.1"
```

Comment:
This should be an [`optionalDependency`](https://yarnpkg.com/lang/en/docs/dependency-types/#toc-optionaldependencies) for those that aren't using this with RN

Reply:
But the only thing that optionalDependencies is good for is enabling an install to not fail if that dependency does not successfully install! One important thing to note here is that it will still attempt to install these optionalDependencies.

On the other hand, regarding peerDependencies, with npm 2, if your package.json listed a peer dependency, when someone installed your package, that dependency was also automatically installed but with npm 3+/yarn installation no peerDependency will actually be installed and instead a warning message about missing peerDependencies will be show.

As such, I believe that moving `react-native-community/async-storage` from peerDependencies to optionalDependencies will actually cause the opposite effect to what your proposition intends.
At worst with `peerDependencies` the non-RN users will see a warning about `redux-persist` missing `@react-native-community/async-storage` peerDependencies.
At worst with `optionalDependency` the non-RN users will end up with `node_modules` that actually includes `@react-native-community/async-storage` package.

Yarn perspective:
[peerDependencies](https://yarnpkg.com/lang/en/docs/dependency-types/#toc-peerdependencies)
[optionalDependencies](https://yarnpkg.com/lang/en/docs/dependency-types/#toc-optionaldependencies)

npm perspective:
[peerDependencies](https://docs.npmjs.com/files/package.json#peerdependencies)
[optionalDependencies](https://docs.npmjs.com/files/package.json#optionaldependencies)

related articles:
[common-npm-mistakes](https://medium.com/@jacob.h.page/common-npm-mistakes-51bf8989079f)
[understanding-the-npm-dependency-model](https://lexi-lambda.github.io/blog/2016/08/24/understanding-the-npm-dependency-model/)

Reply:
Maybe it should not be in package dependencies at all because to make it works you still need to link package manually `react-native link @react-native-community/async-storage`.

Instead we can add  info about usage of `@react-native-community/async-storage` to README.md (seems we already have it)

Reply:
I agree this should not be in deps at all. Instead we should remove `index.native.js` and update the readme accordingly.